plugins/importexport/medra/classes/MedraWebservice.inc.php

```diff
@@ -88,6 +107,22 @@ function _doRequest($action, $arg, $attachment = null) {
 					"<med:$action>$arg</med:$action>" .
 				'</SOAP-ENV:Body>' .
 			'</SOAP-ENV:Envelope>';
+		
+		//Rebuild the multipart SOAP message for the deposit in CRossref: the action is 'deposit' instead of upload
+		if($action == 'deposit'){
+		    $soapMessage =
+		    '<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" ' .
+		    'xmlns:med="http://www.medra.org">' .
+		      '<SOAP-ENV:Header/>' .
+		      '<SOAP-ENV:Body>' .
+		          "<med:$action>" .
+		              "<med:accessMode>01</med:accessMode>" .
+		              "<med:language>$arg</med:language>" .
```

Comment:
I think `$arg` needs to be passed through `htmlspecialchars` or similar, to escape any special characters it might contain.

Reply:
(Also probably `key($attachment)` below)

Reply:
I think the $arg doesn't need to be escape because it will usually be language of tree types: "en_EN, it_IT, de_DE"

Reply:
I applied for the attachment the htmlspecialchars